src/lib/features/client-feature-toggles/delta/delta-cache.test.ts

```diff
@@ -182,4 +182,63 @@ describe('RevisionCache', () => {
             ]),
         );
     });
+
+    it('should not mutate previous feature-updated events when new events with the same feature name are added', () => {
+        const baseEvent: DeltaHydrationEvent = {
+            eventId: 1,
+            features: [
+                {
+                    name: 'base-flag',
+                    type: 'release',
+                    enabled: true,
+                    project: 'streaming-deltas',
+                    stale: false,
+                    strategies: [],
+                    variants: [],
+                    description: null,
+                    impressionData: false,
+                },
+            ],
+            type: 'hydration',
+            segments: [],
+        };
+
+        const deltaCache = new DeltaCache(baseEvent, 10);
+
+        const initialFeatureEvent: DeltaEvent = {
+            eventId: 129,
+            type: DELTA_EVENT_TYPES.FEATURE_UPDATED,
+            feature: {
+                impressionData: false,
+                enabled: false,
+                name: 'streaming-test',
+                description: null,
+                project: 'streaming-deltas',
+                stale: false,
+                type: 'release',
+                variants: [],
+                strategies: [],
+            },
+        };
+        deltaCache.addEvents([JSON.parse(JSON.stringify(initialFeatureEvent))]);
```

Comment:
is it important for this test? if so I'd explain why

Reply:
Yeah.. in order for the test to fail when the mutation was happening the input needs to be cloned otherwise the reference is kept to the original object in this test and the check doesn't fail because the in place mutation also applies to the referenced object. I'll add a comment.